stacks-signer/src/runloop.rs

```diff
@@ -300,10 +345,177 @@ impl<C: Coordinator> RunLoop<C> {
+                request.message = vote.clone();
+                true
+            }
+            Some(None) => {
+                // We have seen this block before, but we have not agreed to sign it. TODO: ignore it
+                debug!("Received a signature share request for a block we have not validated yet.");
+                false
+            }
+            None => {
+                // We have not seen this block before.
+                // TODO: should probably ignore any messages that are not either sBTC transactons or Nakamoto blocks. Leave now for abitrary message signing
+                debug!("Received a signature share request for an unknown message stream. Signing it as is...");
+                true
+            }
+        }
+    }
+
+    /// Helper function to validate a nonce request, updating its message appropriately.
+    /// Note that if the request is for a block, we will update the request message
+    /// as either a hash indicating a vote no or the signature hash indicating a vote yes
+    /// Returns whether the request is valid or not
+    fn validate_nonce_request(&mut self, request: &mut NonceRequest) -> bool {
+        let mut ptr = &request.message[..];
+        let Some(block) = read_next::<NakamotoBlock, _>(&mut ptr).ok() else {
+            // TODO: we should probably reject requests to sign things that are not blocks or transactions (leave for now to enable testing abitrary signing)
+            warn!("Received a nonce request for an unknown message stream. Signing the nonce request as is.");
+            return true;
+        };
+        let Ok(hash) = block.header.signature_hash() else {
+            debug!("Received a nonce request for a block with an invalid signature hash. Ignore it.");
+            return false;
+        };
+        let mut hash_bytes = hash.0.to_vec();
+        let transactions = &self.transactions;
+        let block_info = self.blocks.entry(hash_bytes.clone()).or_insert(BlockInfo {
+            vote: None,
+            valid: false,
+            block: block.clone(),
+        });
+        // Validate the block contents
+        block_info.valid = Self::validate_block(block_info, transactions);
+        if !block_info.valid {
+            // We don't like this block. Update the request to be across its hash with a byte indicating a vote no.
+            debug!("Updating the request with a block hash with a vote no.");
+            hash_bytes.push(b'n');
+        } else {
+            debug!("The block passed validation. Update the request with the signature hash.");
+        }
+        // Cache our vote
+        block_info.vote = Some(hash_bytes.clone());
+        request.message = hash_bytes;
+        true
+    }
+
+    /// Helper function to validate a block's contents
+    fn validate_block(block_info: &BlockInfo, transactions: &[Txid]) -> bool {
```

Comment:
Strictly speaking, we're not validating the block here; we're just checking that a set of required transactions is present.

Reply:
We are validating it as it checks if we received the endpoint response. If not, it will return false.